### PR TITLE
Improved components

### PR DESCRIPTION
**Pull request by Fritsch-Tech**
#### 🔧 Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Codebase improvement
- [ ] Other (if none of the other choices apply)

#### 🚨 Checklist

- [x] I've read the [guidelines for contributing](https://github.com/gremo/react-directus/blob/main/CONTRIBUTING.md)
- [x] I've added necessary documentation (if appropriate)
- [x] I've ensured that my code additions do not fail linting or unit tests (if applicable)

#### Description

This pull request introduces the following changes:

1. **Added All Possible Image Params to Directus Image Component**: I have extended the Directus Image component to include all possible image parameters, providing more flexibility and customization options for users when rendering images.

2. **Authentication Capabilities**: Authentication capabilities have been implemented to allow users to properly access assets when logged in. With this enhancement, authenticated users will have appropriate access permissions to view and interact with assets based on their roles and permissions. Resolving #182

3. **Removed the Need to Get Components from Hook**: While preserving compatibility with existing code, I have modified the codebase to remove the requirement to get components from a hook. This change improves the overall code structure and maintainability.

4. **Updated the README**: The README file has been updated to reflect the recent changes, providing clear instructions on using the updated components and authentication features.

5. **Updated the Test Suite**: I have ensured that the test suite is compatible with the new changes, and all relevant tests have been updated accordingly.

#### Theoretically Minor Edge Cases

The following edge cases may arise, potentially causing minor disruptions in functionality:

- The components may output "unknown" for the `url` prop on first render. This behavior is caused by the necessity to await the Directus token for authentication compatibility. However, this edge case should not significantly impact the overall functionality of the application.

#### Circular Dependency

This pull request introduces a circular dependency as a result of the backwards compatibility implemented for both the DirectusImage and DirectusAsset in the useDirectus hook. The circular dependency may lead to potential issues, including reduced code modularity and maintainability.

### CODE REVIEW

**Comment by gremo** [comment]
Thanks, this is huge. I'll take some time to review.

File: README.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ export const TodoList = () => {
 
 The hook exports a few components for working with Direcuts files [file access](https://docs.directus.io/reference/files/). They are all configured for using the `apiUrl` specified in the provider. Hopefully, more will come in the future 🤗.
 
-> All components, when imported from `react-directus` directly (i.e. not imported using the hook `useDirectus`), can be used in a "standalone" way. It means that they are not bound to the `apiUrl` specified in the provider. In that case, they both accept an `apiUrl` prop.
+> All components, can be used in a "standalone" way. It means that they are not bound to the `apiUrl` specified in the provider. In that case, they both require an `apiUrl` and an optional `accsessToken` prop.
```
Comment (gremo) [inline README.md]: little typo

File: README.md [hunk 2]
```diff
@@ -108,22 +106,26 @@ export const TodoItem = ({ item }) => {
 
 ### `<DirectusImage>`
 
-Computes the URL of the given resource `asset`, rendering it using the `render` prop:
+Computes the URL of the given resource `asset`, rendering it using the `render` prop.
+Implements all directus Transformatrions, as of the time of writing. For further information, see the [Directus documentation](https://docs.directus.io/reference/files/#requesting-a-thumbnail).
```
Comment (gremo) [inline README.md:110]: little typo

**Comment by gremo** [comment]
Can you split this PR in 3 PRs? Like:

- Removed the Need to Get Components from Hook
- Authentication Capabilities
- Change of DirectusImage

I think it contains way too more changes to let me review it and double check!

**Comment by Fritsch-Tech** [comment]
Yes I can do that.
In this case I will probably split it into 4 Seperate PR, the 3 you mentioned and one additional one for the changes in the Provider.

**Comment by gremo** [comment]
Nice, waiting for you then! In the meanwhile I've rebased/fixed all remaining PRs. The renovate PR still throws, I really don't know how to fix it and why it throws.

**Comment by Fritsch-Tech** [comment]
After the merge of the full range of Directus image props, only the authentication capabilities are left.
After that gets merged, would you be willing to consider a update to the npm package?
